pkg/kubelet/cadvisor/cadvisor_linux.go

```diff
@@ -117,6 +124,33 @@ func New(port uint, runtime string) (Interface, error) {
 	return cadvisorClient, nil
 }
 
+func generateCollectorHttpClient(cAdvisorCollectorClientCertFile, cAdvisorCollectorClientPrivateKeyFile string) (*http.Client, error) {
+	//Enable accessing insecure endpoints. We should be able to access metrics from any endpoint
+	tlsConfig := &tls.Config{
+		InsecureSkipVerify: true,
```

Comment:
We also had this discussion when this got merged into cAdvisor (https://github.com/google/cadvisor/pull/1379#issuecomment-233458192). Note that currently cAdvisor also does this (https://github.com/google/cadvisor/blob/master/cadvisor.go#L199).

The comment listed above should go into more details. Essentially I couldn't figure out having this option could decrease the security of how metrics are gathered or stored. There is also the issue where we read metric endpoint via the ip address of the pods, and there really isn't a guarantee that the endpoint's ssl certificate would be configured with this ip address.

I am open to discuss this more and to figure out a more appropriate solution if we need to handle this in another manner.